tests/ui/single_element_loop.rs

```diff
@@ -54,4 +56,19 @@ fn main() {
             }
         };
     }
+
+    // Should lint with correct suggestion (issue #12782)
+    let res_void: Result<bool, bool> = Ok(true);
+    for (Ok(mut _x) | Err(mut _x)) in [res_void] {
+        let ptr: *const bool = std::ptr::null();
+    }
+
+    // Same but with never type.
+    #[derive(Copy, Clone)]
+    enum Void {}
+
+    let res_void: Result<bool, Void> = Ok(true);
+    for (Ok(mut _x) | Err(!)) in [res_void] {
+        let ptr: *const bool = std::ptr::null();
+    }
```

Comment:
Let's just go with the first test case since the second one behaves the same, saves some rustc side work whenever the feature changes

Reply:
Removed the second commit so this test has been removed.